SU2_CFD/src/drivers/CDiscAdjMultizoneDriver.cpp

```diff
@@ -768,8 +821,17 @@ void CDiscAdjMultizoneDriver::ComputeAdjoints(unsigned short iZone) {
 
   AD::ComputeAdjoint(enter_izone, leave_izone);
 
-  AD::ComputeAdjoint(TRANSFER, OBJECTIVE_FUNCTION);
+  /*--- Compute adjoints of transfer and mesh deformation routines,
+   *    only needed on the last inner iteration. ---*/
+
+  if (eval_transfer)
+    AD::ComputeAdjoint(TRANSFER, OBJECTIVE_FUNCTION);
+
+  /*--- Adjoints of dependencies, needed if derivatives of variables
+   *    are extracted (e.g. AoA, Mach, etc.) ---*/
+
```

Comment:
This is an efficiency tweak for cases with mesh deformation, for which we do not need to compute adjoints on every inner iteration.